### PR TITLE
Fourier transform in angular frequency.

### PR DESCRIPTION
**Pull request by AlexanderMisel**
I modified and tested the files.

### CODE REVIEW

File: inst/@sym/fourier.m [hunk 1]
```diff
@@ -17,28 +17,55 @@
...
+%% @seealso{ifourier,laplace}
 %% @end deftypefn
 
 %% Author: Colin B. Macdonald
 %% Keywords: symbolic
 
-function F = fourier(f, x, k)
+function F = fourier(varargin)
+
+  syms f x w;
```
Comment (cbm755) [inline inst/@sym/fourier.m]: remove this line?

File: inst/@sym/fourier.m [hunk 2]
```diff
@@ -17,28 +17,55 @@
...
+  if (nargin == 1)
+    f=varargin{1};
+    x=symvar(f,1);
```
Comment (cbm755) [inline inst/@sym/fourier.m]: I prefer `x = symvar(f, 1)` (just differing whitespace), probably not a big deal but would match the rest of the code

File: inst/@sym/fourier.m [hunk 3]
```diff
@@ -17,28 +17,55 @@
...
+    f=varargin{1};
+    x=varargin{2}; 
+    w=varargin{3};
 
-  cmd = { 'd = sp.fourier_transform(*_ins)'
-          'return d,' };
+  else
+    error('Wrong number of input arguments') 
+ 
+  endif
```
Comment (cbm755) [inline inst/@sym/fourier.m]: for matlab compatibility, please just use `end`.

File: inst/@sym/fourier.m [hunk 4]
```diff
@@ -17,28 +17,55 @@
...
 
-  F = python_cmd (cmd, sym(f), sym(x), sym(k));
+  cmd = { 'sp.FourierTransform._b = -1'
```
Comment (cbm755) [inline inst/@sym/fourier.m]: what is this for?  I cannot find it in docs?  If its important, can you leave a comment here?

File: inst/@sym/ifourier.m [hunk 5]
```diff
@@ -17,31 +17,63 @@
...
+%% @seealso{fourier,laplace}
 %% @end deftypefn
 
 %% Author: Colin B. Macdonald
 %% Keywords: symbolic
 
-function f = ifourier(F, k, x)
+function f = ifourier(varargin)
+
+  syms F w x;
```
Comment (cbm755) [inline inst/@sym/ifourier.m]: as above, remove this

File: inst/@sym/ifourier.m [hunk 6]
```diff
@@ -17,31 +17,63 @@
...
-  cmd = { 'd = sp.inverse_fourier_transform(*_ins)'
-          'return d,' };
+  else
+    error('Wrong number of input arguments') 
+ 
+  endif
 
-  f = python_cmd (cmd, sym(F), sym(k), sym(x));
+  cmd = { 'sp.InverseFourierTransform._a = 0.5/S.Pi'
+          'sp.InverseFourierTransform._b = 1'
```
Comment (cbm755) [inline inst/@sym/ifourier.m]: again please comment this.  Is it really necessary for us to mess around with `_` variables?  Is this some functionality not exposed in the `.inverse_fourier_transform` cmd?

File: inst/@sym/ifourier.m [hunk 7]
```diff
@@ -17,31 +17,63 @@
...
 
 
-%! % tests in fourier.m
-%!assert(true)
+%!test
+%! syms x k
+%! f = exp(-x^2);
+%! F = fourier(f,x,k);
+%! g = ifourier(F,k,x);
+%! assert(isequal(f,g))
```
Comment (cbm755) [inline inst/@sym/ifourier.m]: is just copy-paste of the fourier.m ones?  How about testing that the calling forms without x,k work?  Actually I guess the doctests will get that as you've added it to the docs, so not necessary.  This is ok.

**Comment by cbm755** [comment]
This looks pretty good thanks!  Can you adjust as per my comments about and then I'll merge.

**Comment by cbm755** [comment]
For next time: you should do your work on a branch other than master...   you may have multiple branches submitted for PRs: its likely you'll want/need to revise those, easier if each PR is a new branch (rather than master).

**Comment by AlexanderMisel** [comment]
@cbm755 Maybe you didn't see my comments when I made my first commit, I copy and paste it here:

MATLAB uses non-unitary angular frequency convention, while SymPy uses unitary ordinary frequency.
https://en.wikipedia.org/wiki/Fourier_transform#Other_conventions

Fourier transform has many conventions, which differs only in the choice of `_a` and `_b` of the function called `_fourier_transform` which defined at `class FourierTransform` and `class InverseFourierTranform` in SymPy source, which chose 

<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=default">
F(k) = \int_{-\infty}^\infty f(x) e^{-2\pi i x k} \mathrm{d} x.
</script>

and the inverse

<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=default">
f(x) = \int_{-\infty}^\infty F(k) e^{2\pi i x k} \mathrm{d} k.
</script>

**Comment by AlexanderMisel** [comment]
Actually I am new to python, and all I thought was that `sp` is just an object, and changing its `_a` and `_b` value will not make effects globally. But I dunno whether I am right or wrong. I used to consider calling `_fourier_tranform` directly and passing both parameters as well as a `name` string to it, while it can be complicated.

**Comment by cbm755** [comment]
Ok, I did figure that was what is was for but the code needs some comments about this.  I'm not entirely comfortable messing around with "hidden" (_) variables: by convention those are private.
1.  Is there a way to do this with some "kwarg" in SymPy?  If not, is there an easy patch we could submit to SymPy to add this?  There might be an upstream SymPy issue about this already...
2.  We can always merge it here as you have it, provided it has some warning in comments.
3.  Please add some tests, either in the documentation or as "%!" tests that demonstrate this choice of constants.  Easiest to do it in the documentation, that way we test it (with doctests) and document it.  
4.  We should document this choice.  I think maybe a better reason that "be like Symbolic Math Toolbox" might be for compatibility with the Octave `fft` function.  Can you confirm that is the case?

**Comment by cbm755** [comment]
Here's how to do doctests.  Basically it splits around `@result{}` to find input and output and then tests that it works.
````
@example
@group
syms x
f = fourier(...)
@result{} (sym) x^2
@end group
@end example
````

**Comment by AlexanderMisel** [comment]
@cbm755 I feel difficult to write tests since SymPy calculate symbolic fourier simply by integration and thus some general functions or distributions are not integrated (like Dirac's delta function and Heaviside...). Some inverse fourier transforms I tested today gave the wrong answer and I think it's SymPy's error. Maybe you could help me for the doc.

**Comment by cbm755** [comment]
Sure, I can try to help with docs.

1.  Maybe clean it up as best you can re: my other comments above.  We can always do other PRs for the docs: although it would be good if Travis passes.
2.  I looked a bit at `_fourier_transform`.  Can we call that and pass in `a` and `b` values instead of changing `_a` in the class?

**Comment by AlexanderMisel** [comment]
Certainly we can call it with the following command for Fourier transform if you like:
```
  cmd = { 'from sympy.integrals.transforms import _fourier_transform'
          "F = _fourier_transform(*(_ins+[1,-1,'Fourier']))"
          'return F,'};
```

**Comment by cbm755** [comment]
I think that `cmd` looks better than before.

Re: fft: I mean, fft also has choice and Octave uses 1/N in the inverse.  So presumably some symmetry there if we put the `1/(2*pi)` with the inverse right?

See Travis output for doctest failure, you need `2*sym(pi)` somewhere.

**Comment by cbm755** [comment]
The failing test about execfile is my fault: ignore that, I'll fix it in master.  But the tests for fourier and ifourier are failing...

File: inst/@sym/fourier.m [hunk 8]
```diff
@@ -17,28 +17,76 @@
...
-  cmd = { 'd = sp.fourier_transform(*_ins)'
-          'return d,' };
+  else
+    error('Wrong number of input arguments') 
+ 
+  endif
 
-  F = python_cmd (cmd, sym(f), sym(x), sym(k));
+  cmd = { 'from sympy.integrals.transforms import _fourier_transform'
+          "F = _fourier_transform(*(_ins+[1,-1,'Fourier']))"
```
Comment (cbm755) [inline inst/@sym/fourier.m]: For matlab compat, I prefer:
````
 'F = _fourier_transform(*(_ins+[1,-1,"Fourier"]))'
````

**Comment by cbm755** [comment]
There are also still various comments I made earlier to be addressed.  Let me know if you need help.

**Comment by cbm755** [comment]
I think these changes are done in a slightly different way by @maprieto. I've rebased your documentation updates in #247, which I'll merge soon.

Thanks for this, and please let me know if I've missed something here.

**Pull request closed without merging**